src/main/java/vct/col/veymont/TerminationCheck.scala

```diff
@@ -0,0 +1,199 @@
+      case ASTSpecial.Kind.Fork => Warning("Session Warning: Fork" + deadlockWarning)
+      case ASTSpecial.Kind.Join => Warning("Session Warning: Join" + deadlockWarning)
+      case ASTSpecial.Kind.Lock => Warning("Session Warning: Lock" + deadlockWarning)
+      case ASTSpecial.Kind.Unlock => Warning("Session Warning: Unlock" + deadlockWarning)
+      case ASTSpecial.Kind.Send => Warning("Session Warning: Send" + deadlockWarning)
+      case ASTSpecial.Kind.Recv => Warning("Session Warning: Recv" + deadlockWarning)
+      case _ => super.visit(s)
+    }
+  }
+/*
+  private def isForLoop(l : LoopStatement) : Boolean = {
+    val guardOp = l.getEntryGuard.asInstanceOf[OperatorExpression]
+    val initval = l.getInitBlock.asInstanceOf[BlockStatement].getStatement(0).asInstanceOf[VariableDeclaration].get().head.asInstanceOf[DeclarationStatement].init.get.asInstanceOf[ConstantExpression]
+
+    val updateOp = l.getUpdateBlock.asInstanceOf[BlockStatement].getStatement(0)
+    //TODO: get Value from initval
+    if(isLengthExpr(guardOp.arg(1)))
+      initval.value == 0 && Set(StandardOperator.LT).contains(guardOp.operator) && updateOp.operator == StandardOperator.PostIncr
+    else initval.value < guardOp.arg(1).asInstanceOf[ConstantExpression].value && updateOp.operator == StandardOperator.PostIncr ||
+      initval.value > guardOp.arg(1).asInstanceOf[ConstantExpression].value && updateOp.operator == StandardOperator.PostDecr
+  }
+*/
+  private def isLengthExpr(e : ASTNode) : Boolean = e match {
+    case d : Dereference => d.field == Dereference.ArrayLength
+    case _ => false
+  }
+
+  //TODO: check declaration has int type
+  private def isConstantInitDecl(i : ASTNode) : Boolean =
+    i match {
+      case ib : BlockStatement =>
+        ib.size == 1 && (ib.getStatement(0) match {
+          case init : VariableDeclaration =>
+            init.basetype match {
+              case p : PrimitiveType => p.isInteger && init.get().size == 1 && (init.get().head match {
+                case d : DeclarationStatement => d.init match {
+                  case Some(initval) => initval.isInstanceOf[ConstantExpression]
+                  case None => false
+                }
+                case _ => false
+              })
+              case _ => false
+            }
+          case _ => false
+        })
+      case _ => false
+    }
+
+  private def isConstantOpGuard(g : ASTNode, itName : String) =
+    g match {
+      case op : OperatorExpression =>
+        Set(StandardOperator.LT, StandardOperator.LTE,StandardOperator.GT, StandardOperator.GTE).contains(op.operator) &&
+          op.argslength == 2 && isItName(op.arg(0), itName) && isConstantOrLength(op.arg(1))
+      case _ => false
+    }
+
+  private def isConstantOrLength(e : ASTNode) : Boolean = e match {
+    case c : ConstantExpression => true
+    case d : Dereference => d.field == Dereference.ArrayLength
```

Comment:
and d.object is an array

Reply:
I removed this code because it was not used.